app/scripts/lib/util.ts

```diff
@@ -479,3 +479,58 @@ export function getConversionRatesForNativeAsset({
 
   return conversionRateResult;
 }
+
+/**
+ * Extracts the domain from an RPC endpoint URL
+ *
+ * @param rpcUrl - The RPC endpoint URL
+ * @returns The domain value to track in metrics
+ */
+export function extractRpcDomain(rpcUrl: string) {
+  if (!rpcUrl) {
+    return 'unknown';
+  }
+
+  try {
+    // Try to parse the URL
+    let url;
+    try {
+      url = new URL(rpcUrl);
+    } catch (e) {
+      // Handle URLs without protocol by adding one
+      if (!rpcUrl.startsWith('http://') && !rpcUrl.startsWith('https://')) {
+        try {
+          url = new URL(`https://${rpcUrl}`);
+        } catch (e2) {
+          return 'invalid_url';
+        }
+      } else {
+        return 'invalid_url';
+      }
+    }
+
+    // Extract hostname
+    const { hostname } = url;
+
+    // Handle localhost (including standard localhost IPs)
+    if (
+      hostname === 'localhost' ||
+      hostname === '127.0.0.1' ||
+      hostname.startsWith('192.168.')
+    ) {
+      return 'localhost';
+    }
+
+    // Check for private IP addresses using regex
+    const ipRegex = /^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/u;
+    if (ipRegex.test(hostname)) {
+      return 'private_url';
```

Comment:
I might argue that these represent `ip_address` and not `private_url`, since `url` is a broad term. I don't know if you can change the name or not, but if you can, it might be nice in the future if it didn't use `url`. just a thought.

Reply:
I can certainly check on changing the name. I will bring that up. I will check with those defining the req's